src/wallet/rpcwallet.cpp

```diff
@@ -4183,15 +4304,19 @@ UniValue z_getbalanceforviewingkey(const UniValue& params, bool fHelp)
     if (!EnsureWalletIsAvailable(fHelp))
         return NullUniValue;
 
-    if (fHelp || params.size() < 1 || params.size() > 2)
+    if (fHelp || params.size() < 1 || params.size() > 3)
         throw runtime_error(
-            "z_getbalanceforviewingkey \"fvk\" ( minconf )\n"
+            "z_getbalanceforviewingkey \"fvk\" ( minconf asOfHeight )\n"
             "\nReturns the balance viewable by a full viewing key known to the node's wallet"
             "\nfor each value pool. Sprout viewing keys may be used only if the wallet controls"
             "\nthe corresponding spending key."
             "\nArguments:\n"
             "1. \"fvk\"        (string) The selected full viewing key.\n"
             "2. minconf      (numeric, optional, default=1) Only include transactions confirmed at least this many times.\n"
+            "3. asOfHeight       (numeric, optional) Execute the query as if it were run when\n"
+            "                    the blockchain was at the height specified by this argument.\n"
+            "                    The default is to use the entire blockchain that the node is\n"
+            "                    aware of.\n"
```

Comment:
```suggestion
            "3. asOfHeight   (numeric, optional) Execute the query as if it were run when\n"
            "                the blockchain was at the height specified by this argument.\n"
            "                The default is to use the entire blockchain that the node is\n"
            "                aware of.\n"
```
Document what happens if `asOfHeight` is greater than the height of the last known block.

Reply:
This is now defined in a centralized place, so the reformatting is no longer relevant.